packages/dd-trace/src/appsec/index.js

```diff
@@ -146,6 +145,19 @@ function onRequestQueryParsed ({ req, res, abortController }) {
   handleResults(results, req, res, rootSpan, abortController)
 }
 
+function onRequestCookieParser ({ req, res, abortController, cookies }) {
+  const rootSpan = web.root(req)
+  if (!rootSpan) return
+
+  if (!cookies || typeof cookies !== 'object') return
+
+  const results = waf.run({
+    [addresses.HTTP_INCOMING_COOKIES]: cookies
```

Comment:
You are changing the sintax of the object that is sent to waf.
it was:
```
{
  "server.request.cookies": {
    "cookieName": ["cookieValue"]
  }
}
```

And now it is like:
```
{
  "server.request.cookies": {
    "cookieName": "cookieValue"
  }
}
```

You have remove the array (same in line [112](https://github.com/DataDog/dd-trace-js/pull/3346/files#diff-bfb3e447c35a0c6f6b4131ecef93afce24a08ef37f11fe6587c74be9c86c9920R112))

Reply:
This is intentional yes, I checked with Emile and it's not needed anymore